### ISSUE TITLE
Deployment needs to be possible without tying together all Portlets

### ISSUE THREAD

**Issue by enver-haase** [vaadin/portlet#178]
With vaadin-portlet-1.0.0.beta3, we have this static files bundle with shared resources.
While I do understand that use of Web Components force us to have only one implementation of those per page, and that these should be shared across the portlets, this poses an unnecessary problem the way it is done.
Because, if I develop a new Portlet even with the same version of Vaadin and all, and I have a NewPortletMainView there, then this would become a Web Component with references going into a new static-components-bundle that needs all the portlets in order to be recompiled.

This makes the expected "I can deploy a new Portlet any time" workflow impossible.

It must be possible to develop a new Portlet without source code access to the other Portlets shown on the page.

**Comment by pleku** [comment]
>Because, if I develop a new Portlet even with the same version of Vaadin and all, and I have a NewPortletMainView there, then this would become a Web Component with references going into a new static-components-bundle that needs all the portlets in order to be recompiled.

This is not true - not all the other portlets need to be recompiled. Only the new portlet & static assets bundle. Which is still not perfect though.

>It must be possible to develop a new Portlet without source code access to the other Portlets shown on the page.

Currently it is not necessary to access the other portlets, but the static bundle needs all the portlets due to the "exported web component". The other portlets need to know where to get this static bundle file in the portal, and it should not change during the builds:
```
{
  "externalStatsUrl": "/myproject-bundle/VAADIN/config/stats.json"
}
```

So this issue doesn't seem right. But to improve the situation further, I think there are two options that can try to address the issue of not having to even depend on the static-bundle in every portlet:
1) by using a custom webpack plugin that will exclude any "generic frontend dependencies" like all vaadin components from the portlets and have every portlet create their own bundle with the web component dependency. This is used in the microfrontend labs demo, but it has some other nuances.
2) Trying to utilize the new Webpack 5 module federation feature, which does the same as 1) but "officially" and probably less-error prone.

**Comment by ScriptedAlchemy** [comment]
Federation also has failsafes, fallbacks, and supports urls. The next release adds the ability for modules to independently share vendor code between each other. We also support versioning and semver sharing,SSR,AutomaticVendorFederation, tree-shake support - within module federation. To date I’ve not come across a robust enough system to support my needs - which is why i wrote MF. Let me know if you’ve got any questions

**Comment by pleku** [comment]
Related to #116 vaadin/flow#5537 vaadin/flow#8757

**Comment by roastedcpu** [comment]
I've been analysing the issue and I don't think there is an easy way of achieving the desired result.

One of the main issues is that, when using web components and portlets, multiple bundles will result in duplicate dependencies registrations. Bundling multiple projects, besides being very hard, would result in versions conflicts if multiple bundles use different versions of the same dependency. And maybe bundling would have to be done in deploy-time.

Module federation could work if there was a service exposing the federated modules independently, but it doesn't solve the problem of having completely independent projects when deploying.
